### PR TITLE
Feature/ux enhancements - Resolved #51

### LINKED ISSUE #51: Enhancing UX: Adding Hero Section, Custom Logo, & Possible Navigation
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

Yes, the feature request is related to improving the user experience by adding essential elements that are currently missing to develop additional trust/usability of TrendTrack. Specifically, the absence of a hero section, custom logo, and navigation (if needed) impacts how much trust the user has and usability of the website. These changes should make it more engaging for users and you as the product owner.

### Proposed Solution

The proposed solution involves:
1. Implementing a visually appealing hero section that captures user attention
2. Integrating a customizable logo to enhance branding
3. Adding or refining the navigation to improve the overall user flow and accessibility (optional because of how lean the app is)
4. 
5. This will create a more cohesive and user-friendly interface, making it easier for users to engage with the site and find relevant information.

### Screenshots

TrendTrack Changes

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this (issue will not be assigned if this is skipped)

**Hero Section:**
-Create a reusable Hero component that includes a title, description, and CTA button.
-Use Tailwind CSS for responsive design, ensuring the section is visually appealing on all screen sizes.

**Custom Logo:**
-Add a Logo component that accepts a customizable logo image file.
- Place the logo in the site's header component, ensuring it scales properly using Tailwind’s utility classes like h-12 or w-12 for responsive sizing.

**Navigation (optional)**
-Implement a Navigation bar at the top using React components.
-Use Tailwind’s flex and space-x-* utilities to position links for a clean and responsive layout.

### PR DESCRIPTION
**Pull request by clydehenry3**
### Summary of Changes:
This update introduces a new hero section with custom content and ux/design improvements. The navigation bar has been updated to include the logo, and a custom hero image has been added to enhance the visual appeal of the landing page.

### Related Issue:
This change addresses [Issue #51 ], which requested the addition of a hero section and logo placement in the navigation bar. _**Please see Line 26 in src/components/Hero.tsx to link sections together. Attempted to add a smooth scroll but it kept breaking the file.**_

### Motivation and Context:
The purpose of these updates is to improve the user interface by making the landing page more visually engaging and providing clear branding through the logo and custom imagery in the hero section.

### Dependencies:
No new dependencies were added as part of this update. Tailwind CSS was used to style the components.

Fixes #(issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable):
TrendTrack After

### CODE REVIEW

**Comment by clydehenry3** [comment]
@Bashamega Testing worked great on my machine. Is there anything within my code to improve it? Thanks.

**Comment by Bashamega** [comment]
> @Bashamega Testing worked great on my machine. Is there anything within my code to improve it? Thanks.

The error is with linter:
```
22:73  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
```

File: src/assets/Logo.svg
Comment (Bashamega) [inline src/assets/Logo.svg:1]: use the second icon

File: src/components/Hero.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        <div className="flex-1 pe-16">
+          <h1 className="text-3xl md:text-5xl font-bold">
+            Stay Ahead of GitHub Trends
+          </h1>
+          <p className="py-8 text-sm md:text-base">
+            Discover the top trending repositories with TrendTrack, the ultimate
+            platform for developers, tech enthusiasts, and curious minds.
+            Explore fresh ideas, cutting-edge tools, and innovative projects—all
+            in one user-friendly place. Stay u`pdated daily and see what's
```
Comment (Bashamega) [inline src/components/Hero.tsx:22]: fix linter and fix spelling
```suggestion
            in one user-friendly place. Stay updated daily and see what&apos;s
```

**Review by Bashamega — CHANGES_REQUESTED** [review]

File: src/assets/Logo.svg
Comment (Bashamega) [inline src/assets/Logo.svg:1]: I mean this icon: 
image

File: src/components/Nav.tsx [hunk 2]
```diff
@@ -13,9 +15,12 @@ const Navbar: React.FC = () => {
   return (
     <nav className="bg-gray-800 p-4 w-screen">
```
Comment (Bashamega) [inline src/components/Nav.tsx:16]:
```suggestion
    <nav className="bg-gray-800 p-4">
```

**Review by Bashamega — CHANGES_REQUESTED** [review]

**Comment by Bashamega** [comment]
Also please fix the responsivness

**Comment by clydehenry3** [comment]
> Also please fix the responsivness

I believe the responsive issue is within the cards or the viewport themselves. When using the dev tools and making the screen smaller, it responds great. However when using just the browser, it appears to be within the cards being too wide.

**Comment by clydehenry3** [comment]
I have made the changes of spelling, symbols, and logo.

**Review by Bashamega — APPROVED** [review]
Great work !

**Comment by clydehenry3** [comment]
I'm also going to upload the hero image because when looking on my phone, it looks blurry even though it's an SVG. Going to see how I can improve this.

**Pull request merged**